src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotation.java

```diff
@@ -0,0 +1,446 @@
+package org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode;
+
+import org.broadinstitute.hellbender.tools.funcotator.Funcotation;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A class to represent a Functional Annotation.
+ * Created by jonn on 8/22/17.
+ */
+public class GencodeFuncotation extends Funcotation {
+
+    private static final String FIELD_DELIMITER = "|";
+    private static final String OTHER_TRANSCRIPT_DELIMITER = ";";
+
+    //==================================================================================================================
+
+    private String                  hugoSymbol;                         // TRIVIAL (i.e. by the time we match to a transcript, we have this info regardless to where in the transcript the variant lies)
+    private String                  ncbiBuild;                          // TRIVIAL
+    private String                  chromosome;                         // TRIVIAL
+    private int                     start;                              // TRIVIAL
+    private int                     end;                                // TRIVIAL
+    private VariantClassification   variantClassification;              //          CDS / UTR / INTRON / IGR
+    private VariantClassification   secondaryVariantClassification;     //          CDS / INTRON
+    private VariantType             variantType;                        // TRIVIAL
+    private String                  refAllele;                          // TRIVIAL
```

Comment:
Are yo usure you don't want to use the `Allele` class?

Reply:
I could change it to using the allele class - I don't have a strong opinion on it.